beaker/application.py

```diff
@@ -1,418 +1,963 @@
-
-                if handler_config.is_create():
-                    all_creates.append(static_attr)
-                if handler_config.is_update():
-                    all_updates.append(static_attr)
-                if handler_config.is_delete():
-                    all_deletes.append(static_attr)
-                if handler_config.is_opt_in():
-                    all_opt_ins.append(static_attr)
-                if handler_config.is_close_out():
-                    all_close_outs.append(static_attr)
-
-                self.methods[name] = (abi_meth, handler_config.method_config)
-                self.hints[name] = handler_config.hints()
-
-            # Internal subroutines
-            elif handler_config.subroutine is not None:
-                if handler_config.referenced_self:
-                    setattr(self, name, handler_config.subroutine(bound_attr))
-                else:
-                    setattr(
-                        self.__class__,
-                        name,
-                        handler_config.subroutine(static_attr),
+    def deregister_abi_method(
+        self,
+        method_signature_or_reference: str | ABIReturnSubroutine,
+        /,
+    ) -> None:
+        if isinstance(method_signature_or_reference, str):
+            self.abi_externals.pop(method_signature_or_reference)
+        else:
+            self.abi_externals.pop(method_signature_or_reference.method_signature())
+
+    def _register_bare_external(
+        self,
+        sub: SubroutineFnWrapper,
+        *,
+        actions: MethodConfigDict,
+        override: bool | None,
+    ) -> None:
+        assert all(cc != CallConfig.NEVER for cc in actions.values())
+        for for_action, call_config in actions.items():
+            existing_action = self.bare_actions.get(for_action)
+            if existing_action is None:
+                if override is True:
+                    raise ValueError("override=True, but nothing to override")
+            else:
+                if override is False:
+                    raise ValueError(
+                        f"override=False, but bare external for {for_action} already exists."
                     )
+                assert isinstance(existing_action.action, SubroutineFnWrapper)
+                self.deregister_bare_method(existing_action.action)
+            self.bare_actions[for_action] = OnCompleteAction(
+                action=sub, call_config=call_config
+            )
 
-        self.on_create: Optional[HandlerFunc] = (
-            all_creates.pop() if len(all_creates) == 1 else None
-        )
-        self.on_update: Optional[HandlerFunc] = (
-            all_updates.pop() if len(all_updates) == 1 else None
+    def deregister_bare_method(
+        self,
+        action_name_or_reference: OnCompleteActionName | SubroutineFnWrapper,
+        /,
+    ) -> None:
+        if isinstance(action_name_or_reference, str):
```

Comment:
checking str here instead of OnCompleteActionName? is it not possible to use?

Reply:
Yeah you can't use `isinstance` with `typing.Literal`